packages/block-editor/src/store/selectors.js

```diff
@@ -470,24 +470,22 @@ export const getBlockParents = createSelector(
  *
  * @return {Array} ClientIDs of the parent blocks.
  */
-export const getBlockParentsByBlockName = (
-	state,
-	clientId,
-	blockName,
-	ascending = false
-) => {
-	const parents = getBlockParents( state, clientId, ascending );
-	return map(
-		filter(
-			map( parents, ( id ) => ( {
-				id,
-				name: getBlockName( state, id ),
-			} ) ),
-			{ name: blockName }
-		),
-		( { id } ) => id
-	);
-};
+export const getBlockParentsByBlockName = createSelector(
+	( state, clientId, blockName, ascending = false ) => {
+		const parents = getBlockParents( state, clientId, ascending );
+		return map(
+			filter(
+				map( parents, ( id ) => ( {
+					id,
+					name: getBlockName( state, id ),
+				} ) ),
+				{ name: blockName }
+			),
+			( { id } ) => id
+		);
+	},
+	( state ) => [ state.blocks.parent ]
```

Comment:
If I'm not mistaken we only care about `state.blocks.parent[clientId]`

Reply:
Maybe I misunderstood the dependant references here. I set `state.blocks.parent` because the selector depends on its parent, so if they change, it needs to clear the cache and regenerates.
However, it's already handled by `getBlockParents()` selector. Is this correct?

Reply:
Unless I'm wrong the returned value on change if the parents of a specific block (clientId) change, right?

Reply:
Actually, I'm wrong `state.blocks.parent[ clientId ]` doesn't exist, neighter `state.blocks.parent`. It should be `state.blocks.parents` as we can target all block parents at once.

Reply:
~Beyond that `.parent` doesn't exist 🤦‍♂, is not already handled by the `getBlockParents()` selector?~

oh, got it... we need to be totally specific about the dependant references. Since although it's true that the `getBlockParents()` selector cache depends on `state.blocks.parents`, if this selector depends just on the `clientId`, it's possible that it won't necessarily regenerate its cache if `.parents` tree changes.